bundled/tool/lsp_server.py

```diff
@@ -531,20 +532,22 @@ def references(
 
     pipelines_package = importlib_resources.files(f"{PACKAGE_NAME}.pipelines")
 
-    # Iterate on pipeplines/<pipeline_name>/pipeline.py
+    # Iterate on pipelines/<pipeline_name>/**/*pipeline*.py
     result = []
     for pipeline_dir in pipelines_package.iterdir():
         if not pipeline_dir.is_dir():
             continue
-        pipeline_file = pipeline_dir / "pipeline.py"
-        if not pipeline_file.exists():
-            continue
-
-        # Read the line number and match keywords naively
-        with open(pipeline_file) as f:
-            for i, line in enumerate(f):
-                if word in line:
-                    result.append((pipeline_file, i))
+        # Use glob to find files matching the pattern recursively
+        pipeline_files = glob.glob(f"{pipeline_dir}/**/*pipeline*.py", recursive=True)
+        for pipeline_file in pipeline_files:
+            if not os.path.exists(pipeline_file):
+                continue
```

Comment:
Is this still needed? What is the scenario that a filepath doesn't exist but showing up in `glob`?

Reply:
There could be edge cases where a file is deleted or moved after `glob` has listed it but before the file is opened.

Reply:
This seems rather an edge case? I don't think we need to handle this specifically, since glob isn't taking that long and even if you do a check the file can still be deleted after the check but before it is opened.

Reply:
Yes, agree its an edge case. let me remove this.